internal/commands/app/create.go

```diff
@@ -22,12 +23,47 @@ type CommandCreate struct {
 func (cmd *CommandCreate) Flags(fs *pflag.FlagSet) {
 	fs.StringVar(&cmd.inputs.Project, flagProject, "", flagProjectUsage)
 	fs.StringVarP(&cmd.inputs.Name, flagName, flagNameShort, "", flagNameUsage)
+	fs.StringVarP(&cmd.inputs.From, flagFrom, flagFromShort, "", flagFromUsage)
 	fs.StringVarP(&cmd.inputs.Directory, flagDirectory, flagDirectoryShort, "", flagDirectoryUsage)
 	fs.VarP(&cmd.inputs.DeploymentModel, flagDeploymentModel, flagDeploymentModelShort, flagDeploymentModelUsage)
 	fs.VarP(&cmd.inputs.Location, flagLocation, flagLocationShort, flagLocationUsage)
 	fs.StringVarP(&cmd.inputs.DataSource, flagDataSource, flagDataSourceShort, "", flagDataSourceUsage)
-	// TODO(REALMC-8134): Implement dry-run for app create command
-	// fs.BoolVarP(&cmd.inputs.DryRun, flagDryRun, flagDryRunShort, false, flagDryRunUsage)
+	fs.BoolVarP(&cmd.inputs.DryRun, flagDryRun, flagDryRunShort, false, flagDryRunUsage)
+}
+
+func (cmd *CommandCreate) commandString(omitDryRun bool) string {
+	sb := strings.Builder{}
+	sb.WriteString("app create")
+
+	if cmd.inputs.Project != "" {
+		sb.WriteString(" --project ")
+		sb.WriteString(cmd.inputs.Project)
+	}
+	if cmd.inputs.Name != "" {
+		sb.WriteString(" --name ")
+		sb.WriteString(cmd.inputs.Name)
+	}
+	if cmd.inputs.From != "" {
+		sb.WriteString(" --from ")
+		sb.WriteString(cmd.inputs.From)
+	}
+	if cmd.inputs.Directory != "" {
+		sb.WriteString(" --app-dir ")
+		sb.WriteString(cmd.inputs.Directory)
+	}
+	sb.WriteString(" --deployment-model ")
```

Comment:
any reason we didn't have conditionals around location/deployment model?

Reply:
Those variables are guaranteed as of now to always have a value where this function is called (handler). Can add them if you like.

Reply:
if they are empty/default values, lets leave them out of the printed string, on the condition that leaving them out doesn't then force the user into answering some prompt about which location/deployment model they want

basically, let's print the user the minimally viable command string here, all while not forcing them to hit another prompt